### ISSUE TITLE
[HOW] do i find the issue that crashes kibot in my project

### ISSUE THREAD

**Issue by harm363** [INTI-CMNB/KiBot#307]
I hope i assigned the correct tag, it was as close encounter with bugfix but it didnt feel like a bug. 
I have a project with multiple schematic files, that crashes kibot. i am convinced the issue is in my Kicad files since the (exactly) same kibot and kibot.yaml files run with other multi sheet project with no issues. The issue seems to occur when saving a schematic to the tmp folder when running kibot with the DEFAULT variant. 

what i want to know is how to find the issue from here, since my python is far from good enough to follow what exactly is happening here and why this issue surfaces when a variant is called when running kibot.

I have added the  kibot output to the debug.txt file. 
[debug.txt](https://github.com/INTI-CMNB/KiBot/files/9673887/debug.txt)
the error message is on the end of the file. 

**Do you have some PCB/Schematic to use as example?**
i have narrowed the offending circuit to this one. (note: i am not the developer of this schematic, just the maintainer):
[schematic.zip](https://github.com/INTI-CMNB/KiBot/files/9674144/schematic.zip)

**Do you have some configuration file (.kibot.yaml) that you are using?**
You can attach it or paste the content in the following section:
```yaml
kibot:
  version: 1

import:
  - file: ../source/variant.kibot.yaml
    variants: all

outputs:
  - name: schematic
    type: pdf_sch_print
    comment: generate schematic in pdf
    dir: "."

  - name: PNPpositions
    type: position
    comment: generate position files
    dir: assembly    
    options:
      bottom_negative_x: true
      format: CSV
      only_smd: true
      include_virtual: false
      
  - name: interactiveBOM
    type: ibom
    comment: interactice bom
    dir: "assembly"
    options:
      #output: 'ibom.html'
      highlight_pin1: true
      dark_mode: true
      blacklist: "TP*,FID*,REF**,SYM*,JP*,H*"

  - name: render_3d
    type: render_3d
    dir: "."
    options:
      rotate_x: 2
      width: 3840
      height: 2160 
      zoom: 4

  - name: bom
    type: bom
    dir: 'assembly'
    options:
      format: CSV
      hide_pcb_info: True
      hide_stats_info: True
      #output: 'bom.csv'
      columns:
        - field: 'row'
        - field: 'Value'
        - field: 'References'
        - field:  Quantity Per PCB
        - field: 'Partnumber'
          join:  'MPN'
        - field: 'digikey'
        - field: 'Alternative'
        - field: 'Footprint'
        - field: 'vendor'

#gerbers
  - name: JLCPCB_gerbers
    comment: Gerbers compatible with JLCPCB
    type: gerber
    dir: JLCPCB
    options: &gerber_options
      exclude_edge_layer: true
      exclude_pads_from_silkscreen: true
      plot_sheet_reference: false
      plot_footprint_refs: true
      plot_footprint_values: false
      force_plot_invisible_refs_vals: false
      tent_vias: true
      use_protel_extensions: true
      create_gerber_job_file: false
      disable_aperture_macros: true
      gerber_precision: 4.6
      use_gerber_x2_attributes: false
      use_gerber_net_attributes: false
      line_width: 0.1
      subtract_mask_from_silk: true
      inner_extension_pattern: '.gp%n'
    layers:
      # Note: a more generic approach is to use 'copper' but then the filenames
      # are slightly different.
      - F.Cu
      - B.Cu
      - In1.Cu
      - In2.Cu
      - In3.Cu
      - In4.Cu
      - In5.Cu
      - In6.Cu
      - F.SilkS
      - B.SilkS
      - F.Mask
      - B.Mask
      - B.Paste
      - F.Paste
      - Edge.Cuts

  - name: JLCPCB_drill
    comment: Drill files compatible with JLCPCB
    type: excellon
    dir: JLCPCB
    options:
      pth_and_npth_single_file: false
      pth_id: '-PTH'
      npth_id: '-NPTH'
      metric_units: true
      map: gerber
      route_mode_for_oval_holes: false
      output: "%f%i.%x"

  - name: JLCPCB
    comment: ZIP file for JLCPCB
    type: compress
    #dir: JLCPCB
    options:
      files:
        - from_output: JLCPCB_gerbers
          dest: fabrication/
        - from_output: JLCPCB_drill
          dest: fabrication/


#  - name: datasheets
#    comment: datasheets
#    type: download_datasheets
#    dir: ../datasheets
```
```yaml
kibot:
  version: 1

variants:
  - name: "DEFAULT"
    type: "kibom"
    comment: "default variant"
    file_id: "_DEFAULT"
    variant: "DEFAULT"
```

**Environment (please complete the following information):**
Where are you running KiBot:

- Using a desktop system? Which OS, distro and version?
I use Kibot on a desktop system: linux, Pop!_OS 22.04 LTS 
kibot version; KiBot 1.3.0

**Additional context**
A hardware project has the following layout

- config (for kibot config files) etc
- documentation 
- fabrication (all the kibot outputs end up here)
-   - assembly (for assembly files)
- source (kicad source files)
- Makefile to easy run kibot etc.

**Comment by harm363** [comment]
relevant excerpt form the debug output:
Traceback (most recent call last):
  File "/usr/bin/kibot", line 33, in <module>
    sys.exit(load_entry_point('kibot==1.3.0', 'console_scripts', 'kibot')())
  File "/usr/lib/python3/dist-packages/kibot/__main__.py", line 342, in main
    generate_outputs(outputs, args.target, args.invert_sel, args.skip_pre, args.cli_order, args.no_priority,
  File "/usr/lib/python3/dist-packages/kibot/kiplot.py", line 462, in generate_outputs
    _generate_outputs(outputs, targets, invert, skip_pre, cli_order, no_priority, dont_stop)
  File "/usr/lib/python3/dist-packages/kibot/kiplot.py", line 453, in _generate_outputs
    run_output(out, dont_stop)
  File "/usr/lib/python3/dist-packages/kibot/kiplot.py", line 380, in run_output
    out.run(get_output_dir(out.dir, out))
  File "/usr/lib/python3/dist-packages/kibot/out_base.py", line 162, in run
    self.options.run(self.expand_filename(output_dir, self.options.output))
  File "/usr/lib/python3/dist-packages/kibot/out_any_sch_print.py", line 60, in run
    fname = GS.sch.save_variant(sch_dir)
  File "/usr/lib/python3/dist-packages/kibot/kicad/v6_sch.py", line 1762, in save_variant
    self.save(fname, dest_dir)
  File "/usr/lib/python3/dist-packages/kibot/kicad/v6_sch.py", line 1758, in save
    sch.sch.save(sch.flat_file if cross else sch.file, dest_dir, base_sheet, saved)
  File "/usr/lib/python3/dist-packages/kibot/kicad/v6_sch.py", line 1740, in save
    comp = s.component
AttributeError: 'SymbolInstance' object has no attribute 'component'
make: *** [Makefile:16: kibot] Error 1

**Comment by set-soft** [comment]
Hi @harm363 !
It was a bug. The file you sent was used as a sub-sheet and also as a standalone schematic.
Printing this sheet alone didn't generate problems, but printing it as a sub-sheet did.
This was related to the variants and filters, without using them it always worked.
The problem was related to the fact that the sheet has symbol instances, but they belong to the root sheet. As this sheet was edited standalone KiCad saved the info in the file.
When we stored the sub-sheet we tried to find changes applied by the filters, but the changes are valid only to the root sheet, making the code fail.
The above patch fixes it.
